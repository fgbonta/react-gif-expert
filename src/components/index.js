
//re-export
export * from './AddCategory';
export * from './GifGrid';
export * from './GifItem';