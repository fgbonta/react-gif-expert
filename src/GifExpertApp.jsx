import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

// despues de la version 17 de react no hace falta importarlo
// import React from 'react'

// https://developers.giphy.com/dashboard/?
// fgbonta@gmail.com Agustin43@
// react-app-curso 07yugR7qqyLHpAcR6uj7lRamVCh2JON0

export const GifExpertApp = () => {

    //no puede haber hooks en if (no deben hacer)
    const [categories, setCategories] = useState(['One Punch']);

    const onAddCategory = (value) => {

        /* const [,data2] = ['One Punch','Dragon Ball'];
        const { nameId } = { nameId:'26954', status: true};
        const newData = [...['One Punch','Dragon Ball'],'hola'];
        const newData2 = { ...{ nameId:'26954', status: true},tete:'sisi'};
        console.log(data2,nameId,newData,newData2); */

        if( categories.includes(value) ) return;
        
        /**
         * no se puede usar categories.push('hulk'), no se puede mutar
         */
        setCategories([value,...categories]);
    }

    return (
        <>
            {/** titulo */}
            <h1>GifExpertApp</h1>

            {/** input */}
            <AddCategory 
                //setCategories={ setCategories }
                onNewCategory={ (value)=> onAddCategory(value) }
            />

            {/** listado de Gif */}
            
            { 
                categories.map( (category) => 
                    (
                        <GifGrid 
                            key={ category }
                            category={ category }
                        />
                    )
                ) 
            }

        </>

    )
}
