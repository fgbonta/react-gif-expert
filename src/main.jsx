import React from 'react'
import ReactDOM from 'react-dom/client'
import { GifExpertApp } from './GifExpertApp';

import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  //modo strict se aplica solo en desarrollo
  <React.StrictMode>
    <GifExpertApp />
  </React.StrictMode>
)
