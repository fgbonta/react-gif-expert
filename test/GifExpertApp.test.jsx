import { fireEvent, screen ,render } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => {

    test('Debe hacar match con el SnapShot', () => {

        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();

    });    

    test('Debe poder agregar un categoria inexistente', () => {
                
        const category = "valorant";

        render(<GifExpertApp />);

        const input = screen.getByRole('textbox');
        fireEvent.input(input, { target: { value: category } });

        expect( input.value ).toBe(category);

        const form = screen.getByRole('form',{ name: 'form-add' });
        fireEvent.submit(form);

        expect( screen.getAllByText(category).length ).toBe(1);
        expect( input.value ).toBe("");

    });

    test('No Debe poder agregar un categoria existente', () => {
                
        const category = "dragon ball";

        render(<GifExpertApp />);

        const input = screen.getByRole('textbox');
        fireEvent.input(input, { target: { value: category } });

        expect( input.value ).toBe(category);

        const form = screen.getByRole('form',{ name: 'form-add' });
        fireEvent.submit(form);

        expect( screen.getAllByText(category).length ).toBe(1);
        expect( input.value ).toBe("");

        //repito la carga
        fireEvent.input(input, { target: { value: category } });
        expect( input.value ).toBe(category);
        fireEvent.submit(form);
        expect( input.value ).toBe("");

        expect( screen.getAllByText(category).length ).toBe(1);
        //screen.debug();
    });

});