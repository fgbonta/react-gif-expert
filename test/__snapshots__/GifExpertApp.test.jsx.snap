// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe hacar match con el SnapShot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form-add"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <h2>
    cargando..
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
